packages/core/src/render3/jit/module.ts

```diff
@@ -19,14 +20,63 @@ import {reflectDependencies} from './util';
 
 const EMPTY_ARRAY: Type<any>[] = [];
 
+interface ModuleQueueItem {
+  moduleType: Type<any>;
+  ngModule: NgModule;
+}
+
+const moduleQueue: ModuleQueueItem[] = [];
+
+/**
+ * Enqueues moduleDef to be checked later to see if scope can be set on its
+ * component declarations.
+ */
+function enqueueModuleForDelayedScoping(moduleType: Type<any>, ngModule: NgModule) {
+  moduleQueue.push({moduleType, ngModule});
+}
+
+/**
+ * Loops over queued module definitions, if a given module definition has all of its
+ * declarations resolved, it dequeues that module definition and sets the scope on
+ * its declarations.
+ */
+export function flushModuleScopingQueueAsMuchAsPossible() {
+  for (let i = moduleQueue.length - 1; i--;) {
```

Comment:
```suggestion
  for (let i = moduleQueue.length; --i;) {
```

But that's a bit magic, instead:

```suggestion
  for (let i = moduleQueue.length - 1; i >= 0; i--) {
```

Reply:
Or we can use the well-known `-->` "goes-to" operator 😜
```suggestion
  let i = moduleQueue.length;
  while (i --> 0) {
```

Reply:
Yeah. I saw that and fix it.

I like the "goes to" operator. haha!